learning/step_by_step/scripting.rst

```diff
@@ -7,75 +7,86 @@ Introduction
-for a long time, even inside companies, where game designers wish to
-have more control of the game flow.
-
-Many products have been shipped promising a no-programming environment,
-but the result is often incomplete, too complex or inefficient compared
-to traditional code. As a result, programming is here to stay for a long
-time. In fact, the general direction in game engines has been to add
-tools that try to reduce the amount of code that needs to be written for
-specific tasks, to speed up development.
-
-In that sense, Godot has taken some useful design decisions towards that
-goal. The first and most important is the scene system. The aim of it is
-not obvious at first, but works well later on. That is, to relieve
-programmers from the responsibility of architecting code.
-
-When designing games using the scene system, the whole project is
-fragmented into *complementary* scenes (not individual ones). Scenes
-complement each other, instead of being separate. There will be plenty
-of examples about this later on, but it's very important to remember it.
-
-For those with a good amount of programming expertise, this means a
-different design pattern to MVC. Godot promises efficiency at the
-expense of dropping the MVC habits, which are replaced by the *scenes as
-a complement* pattern.
-
-Godot also uses the `extend <http://c2.com/cgi/wiki?EmbedVsExtend>`__
-pattern for scripting, meaning that scripts extend from all the
-available engine classes.
+without programming. This appeals to many people who would love to be able to
+make a game without having to learn how to code. Furthermore, in many game
+companies, developers write custom tools that give their game designers more
+control over their game's flow. It's tempting, therefore, to imagine that
+someone could could write the ultimate tool that just works for any game type.
+
+Many products have been shipped promising such a no-programming environment,
+but the results often fall short of expectations. The final projects they
+produce regularly end up being too complex or require solutions that are too
+inefficient compared to what could have been accomplished with traditional code.
+
+In contrast, we believe that programming is a critical part of game development
+and is not going anywhere anytime soon. In fact, game engines have been moving
+in this direction, adding tools that try to reduce the amount of code that needs
+to be written rather than eliminating it. The engine can provide many general
+solutions, while the developer can use code to accomplish specific tasks.
+
+Godot has embraced this goal from the beginning, and it has influenced many of
+its design decisions. First and foremost is the scene system. This system has
+many benefits, but fundamentally its goal is to relieve programmers from the
+responsibility of having to implement an overall architecture.
+
+As discussed in the previous section, when designing games using scenes, you
```

Comment:
This is another paragraph that you could potentially remove. I'm not sure what the point is in this section of emphasizing that scenes are complementary, in a tutorial that's about scripting. But I'm happy to leave it in as well -- up to you.